### ISSUE TITLE
Error Build --use_dml

### ISSUE THREAD

**Issue by Armin234** [microsoft/onnxruntime#2761]
Hi,

I'm trying do build the DirectML Provider, but running the commad 
build.bat --config RelWithDebInfo --build_shared_lib --parallel --use_dml

the build process stops with the following error:

...... 
word_conv_embedding.cc
  cpu_contrib_kernels.cc
  element_wise_ops.cc
  unique.cc
  activations.cc
  element_wise_ops.cc
  unique.cc
  LINK : Es wurde ein mit /GL kompiliertes MSIL .netmodule oder module gefunden. Verkn³pfung wird mit /LTCG erneut gestartet. F³gen Sie
  /LTCG zur Linkbefehlszeile hinzu, um die Linkerleistung zu erh÷hen.
  onnxruntime_providers.vcxproj -> C:\Temp\onnxruntime\build\Windows\RelWithDebInfo\RelWithDebInfo\onnxruntime_providers.lib
Traceback (most recent call last):
  File "C:\Temp\onnxruntime\\tools\ci_build\build.py", line 981, in <module>
    sys.exit(main())
  File "C:\Temp\onnxruntime\\tools\ci_build\build.py", line 920, in main
    build_targets(cmake_path, build_dir, configs, args.parallel)
  File "C:\Temp\onnxruntime\\tools\ci_build\build.py", line 415, in build_targets
    run_subprocess(cmd_args)
  File "C:\Temp\onnxruntime\\tools\ci_build\build.py", line 200, in run_subprocess
    completed_process = subprocess.run(args, cwd=cwd, check=True, stdout=stdout, stderr=stderr, env=my_env, shell=shell)
  File "C:\Python37\lib\subprocess.py", line 487, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['C:\\Program Files\\CMake\\bin\\cmake.EXE', '--build', 'C:\\Temp\\onnxruntime\\\\build\\Windows\\RelWithDebInfo', '--config', 'RelWithDebInfo', '--', '/maxcpucount:8']' returned non-zero exit status 1.

Regards
Armin

**Comment by patlevin** [comment]
It would be helpful to at least include the Visual Studio version.
I just ran a build using the same settings and --cmake_generator "Visual Studio 16 2019" and it compiled just fine.
The linker error usually only pops up if there are conflicting settings between the compiler ("whole program optimisation") and the linker ("Link time code generation").
Have you tried a rebuild without the --parallel flag? I ran into issues before when using --parallel, because MSBuild didn't seem to catch all updates in the dependencies.
